app/src/main/kotlin/com/wire/android/notification/CallNotificationManager.kt

```diff
@@ -0,0 +1,160 @@
+package com.wire.android.notification
+
+import android.app.Notification
+import android.content.Context
+import android.os.Build
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.core.app.NotificationChannelCompat
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import com.wire.android.R
+import com.wire.kalium.logic.data.conversation.Conversation
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.feature.call.Call
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@OptIn(
+    ExperimentalAnimationApi::class,
+    ExperimentalMaterial3Api::class,
+    ExperimentalMaterialApi::class,
+    ExperimentalComposeUiApi::class
+)
+@Singleton
+class CallNotificationManager @Inject constructor(private val context: Context) {
+
+    private val notificationManager = NotificationManagerCompat.from(context)
+
+    fun handleNotifications(calls: List<Call>, userId: QualifiedID?) {
+        if (calls.isEmpty() || userId == null) hideCallNotification()
+        else showIncomingCallNotification(calls.first(), userId)
+    }
+
+    fun hideCallNotification() = notificationManager.cancel(NOTIFICATION_ID)
+
+    private fun showIncomingCallNotification(call: Call, userId: QualifiedID) {
+        createNotificationChannelIfNeeded()
+
+        notificationManager.notify(NOTIFICATION_ID, getNotification(call, userId))
+    }
+
+    private fun createNotificationChannelIfNeeded() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val notificationChannel = NotificationChannelCompat
+                .Builder(CHANNEL_ID, NotificationManagerCompat.IMPORTANCE_MAX)
+                .setName(CHANNEL_NAME)
+                .build()
+
+            notificationManager.createNotificationChannel(notificationChannel)
+        }
+    }
+
+    private fun getNotification(call: Call, userId: QualifiedID): Notification {
+        val conversationIdString = call.conversationId.toString()
+        val userIdString = userId.toString()
+        val title = getNotificationTitle(call)
+        val content = getNotificationBody(call)
+
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            getCompatNotification(title, content, conversationIdString, userIdString)
+        } else {
+            getOldNotification(title, content, conversationIdString, userIdString)
```

Comment:
Can't we use `NotificationCompat.Builder` also for versions older than O?

Reply:
that is a good question! I don't know why did I add that `if`, looks like Compat works for both